<launch>
  <arg name="mav_name" default="firefly"/>
  <arg name="world_frame" default="odom" />
  <arg name="verbose" default="false"/>
  <arg name="voxel_size" default="0.20" />
  <arg name="voxels_per_side" default="16" />
  <arg name="robot_radius" default="0.6" />
  <arg name="clear_radius" default="1.0" />

  <group ns="$(arg mav_name)">
    <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr" clear_params="true">
      <remap from="pointcloud" to="vi_sensor/camera_depth/depth/points"/>
      <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
      <remap from="voxblox_node/tsdf_map_out" to="tsdf_map" />

      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="publish_esdf_map" value="true" />
      <param name="publish_pointclouds" value="true" />
      <param name="use_tf_transforms" value="true" />
      <param name="update_mesh_every_n_sec" value="1.0" />
      <param name="clear_sphere_for_planning" value="true" />
      <param name="world_frame" value="$(arg world_frame)" />
      <param name="clear_sphere_radius" value="1.0"/>
      <param name="occupied_sphere_radius" value="0.0"/>
      <param name="verbose" value="false" />
    </node>

  </group>
</launch>
